game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsBuilder.java

```diff
@@ -0,0 +1,71 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+
+/** Build the steps for the fire round (roll dice, select casualties, and mark casualties) */
+@UtilityClass
+public class FireRoundStepsBuilder {
+
+  @Builder
+  public static class Parameters {
+    @NonNull final BattleState battleState;
+    @NonNull final BattleActions battleActions;
+    @NonNull final Function<BattleState, List<FiringGroup>> firingGroupSplitter;
+    @NonNull final BattleState.Side side;
+    @NonNull final MustFightBattle.ReturnFire returnFire;
+    @NonNull final BiFunction<IDelegateBridge, RollDice, DiceRoll> roll;
+    @NonNull final BiFunction<IDelegateBridge, SelectCasualties, CasualtyDetails> selectCasualties;
```

Comment:
nit, as a function name, what do you think of chaning 'selectCasualties' to 'casualtySelector'. As a verb I think it fits a bit better, eg:

```
CasualtyDetails casualtyDetails = selectCasualties.apply(...)
// vs
asualtyDetails casualtyDetails =  casualtySelector.apply( ...)
```

Reply:
Ditto for 'roll', perhaps better as 'diceRoller'?

`firingGroupSplitter` already has a verb kind of name, updating these last two bi-functions I think would make that naming more consistent in this class overall.